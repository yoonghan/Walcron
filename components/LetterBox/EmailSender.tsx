interface EmailSenderProps {
  writeTo: string
  writeFrom: string
}

const EmailSender = ({ writeTo, writeFrom }: EmailSenderProps) => {
  useEffect(() => {
    const subject = encodeURIComponent(`Contact from ${writeFrom} website`)
    const body = "Hello there, "

    window.location.href = `mailto:${writeTo}?subject=${subject}&body=${body}`
  }, [writeFrom, writeTo])

  return (
    <div className="container">
      Apologies that we do require you to use your own mailbox
      <style jsx>{`
        .container {
          padding: 50px;
          background: #fff;
          color: #000;
          border-radius: 3px;
          min-width: 150px;
        }
      `}</style>
    </div>
  )
}

export default EmailSender
