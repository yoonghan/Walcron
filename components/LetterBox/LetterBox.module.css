.container {
    display: flex;
    margin-top: 20px;
    justify-content: center;
}
.input {
    color: var(--link);
    background-color: transparent;
    border: 1px solid var(--link);
    margin: 0;
    min-width: 200px;
    display: block;
    font-size: 1rem;
    box-sizing: content-box;
    padding: 1rem;
    border-radius: 0.3rem;
    height: 1.1rem;
    text-shadow: 0 0 0 #ebebeb;
}