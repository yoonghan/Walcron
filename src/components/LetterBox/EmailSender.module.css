.container {
    padding: 50px;
    background: #fff;
    color: #000;
    border-radius: 3px;
    min-width: 150px;
}