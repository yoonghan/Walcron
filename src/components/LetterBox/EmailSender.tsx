`use strict`

import * as React from "react";

interface EmailSenderProps {
  writeTo: string;
  writeFrom: string;
}

const _openMailSender = (writeTo:string, writeFrom:string) => {

  const subject = encodeURIComponent(`Contact from ${writeFrom}`);
  const body = "Hello there, ";

  window.location.href = `mailto:${writeTo}?subject=${subject}&body=${body}`;
}

const EmailSender: React.SFC<EmailSenderProps> = ({writeTo, writeFrom}) => {

  _openMailSender(writeTo, writeFrom);

  return (<div className="container">
    Using your functioning mailbox.
    <style jsx>{`
      .container {
        padding: 50px;
        background: #FFF;
        color: #000;
        border-radius: 3px;
        min-width: 150px;
      }
    `}</style>
  </div>)
}

export default EmailSender;
