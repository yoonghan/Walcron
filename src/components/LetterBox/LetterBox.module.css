.container {
  display: flex;
  margin-top: 20px;
  justify-content: center;
}
.input {
  background-color: white;
  border: 1px solid var(--border-color);
  margin: 0;
  min-width: 200px;
  display: block;
  font-size: 1rem;
  box-sizing: content-box;
  padding: 1rem;
  border-radius: 0.3rem;
  height: 1.1rem;
}

@media only screen and (max-width: 640px) {
  .input {
    display: none;
  }
}
