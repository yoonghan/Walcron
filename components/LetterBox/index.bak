import Button from "../Button"
import produce, { Draft } from "immer"
import Modal from "../Modal"
import EmailSender from "./EmailSender"
import { LINK, LINK_COLOR } from "../../shared/style"
import { EMAIL } from "../../shared/const"

export interface LetterBoxProps {}

export interface LetterBoxStates {
  isDialogShow: boolean
  name: string
}

class LetterBox extends React.PureComponent<LetterBoxProps, LetterBoxStates> {
  constructor(props: LetterBoxProps) {
    super(props)
    this.state = {
      isDialogShow: false,
      name: "",
    }
  }

  _closeCallback = () => {
    this.setState(
      produce((draft: Draft<LetterBoxStates>) => {
        draft.isDialogShow = false
        draft.name = ""
      })
    )
  }

  _onSubmitPressed = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    this._onClickSendButton()
  }

  _onClickSendButton = () => {
    const { name } = this.state
    if (!name || name.trim() === "") {
      return
    }

    this.setState(
      produce((draft: Draft<LetterBoxStates>) => {
        draft.isDialogShow = true
      })
    )
  }

  _onChangeLetterBoxInput = (event: React.ChangeEvent<HTMLInputElement>) => {
    const input = event.target.value
    this.setState(
      produce((draft: Draft<LetterBoxStates>) => {
        draft.name = input
      })
    )
  }

  _email = () => {
    return EMAIL.replace(/_/g, "")
  }

  _renderNoJs = () => (
    <span>
      Write to us at{" "}
      <a href={`mailto:${this._email()}`} className="container">
        {this._email()}
      </a>
      <style jsx>{`
        .container {
          color: orange;
        }
      `}</style>
    </span>
  )

  render() {
    const { isDialogShow, name } = this.state

    return (
      <div>
        If you are interested to talk to us, leave us your contact. Let us reach
        you instead.
        <form
          className={"letterbox-container"}
          onSubmit={this._onSubmitPressed}
        >
          <input
            type="text"
            autoComplete="off"
            className={"letterbox-input"}
            maxLength={200}
            placeholder={"Honorific and name"}
            onChange={this._onChangeLetterBoxInput}
            value={name}
          />
          <Button onClickCallback={this._onClickSendButton}>Send</Button>
          {isDialogShow && (
            <Modal cancelCallback={this._closeCallback}>
              <EmailSender writeFrom={name.trim()} writeTo={this._email()} />
            </Modal>
          )}
          <style jsx>{`
            .letterbox-container {
              display: flex;
              margin-top: 20px;
              justify-content: center;
            }
            .gap {
              margin-top: 20px;
            }
            .letterbox-input {
              color: ${LINK_COLOR};
              background-color: transparent;
              border: 1px solid ${LINK.FOREGROUND};
              margin: 0;
              min-width: 200px;
              display: block;
              font-size: 1rem;
              box-sizing: content-box;
              padding: 1rem;
              border-radius: 0.3rem;
              height: 1.1rem;
              text-shadow: 0 0 0 #ebebeb;
            }
          `}</style>
        </form>
      </div>
    )
  }
}

export default LetterBox
