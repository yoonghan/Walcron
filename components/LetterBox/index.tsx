import { useState } from "react"
import Button from "../Button"
import Modal from "../Modal"
import EmailSender from "./EmailSender"
import styles from "./LetterBox.module.css"

const LetterBox = () => {
  const [isDialogShown, setIsDialogShown] = useState(false)
  const [name, setName] = useState("")

  const onClose = () => {
    setIsDialogShown(false)
    setName("")
  }

  const onSubmitPressed = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    onSendButtonClick()
  }

  const onSendButtonClick = () => {
    setIsDialogShown(true)
  }

  const email = "walcoor_perati_on@gm_ail.com".replace(/_/g, "")

  return (
    <div>
      If you are interested to talk to us, leave us your contact. Let us reach
      you instead.
      <form className={styles.container} onSubmit={onSubmitPressed}>
        <input
          type="text"
          autoComplete="off"
          className={styles.input}
          maxLength={200}
          placeholder={"Honorific and name"}
          onChange={(event) => setName(event.target.value)}
          value={name}
        />
        <Button>Write To Us</Button>
        {isDialogShown && (
          <Modal onCancel={onClose}>
            <EmailSender writeFrom={name.trim()} writeTo={email} />
          </Modal>
        )}
      </form>
    </div>
  )
}

export default LetterBox
