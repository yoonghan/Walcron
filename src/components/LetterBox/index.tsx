import { useCallback, useState } from "react"
import { createConfirmation } from "react-confirm"
import Button from "../Button"
import Dialog from "../Dialog"
import dialogRootCreator from "../Dialog/dialogRootCreator"
import EmailSender from "./EmailSender"
import styles from "./LetterBox.module.css"

const confirmation = createConfirmation(
  EmailSender,
  1000,
  dialogRootCreator.create()
)

const email = "walcoor_perati_on@gm_ail.com".replace(/_/g, "")

const LetterBox = () => {
  const [name, setName] = useState("")

  const onSubmitPressed = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    onSendButtonClick()
  }

  const onSendButtonClick = useCallback(() => {
    confirmation({
      writeFrom: name.trim(),
      writeTo: email,
      onCancel: () => {
        setName("")
      },
    })
  }, [name])

  return (
    <div>
      If you are interested to talk to us, leave us your contact. Let us reach
      you instead.
      <form className={styles.container} onSubmit={onSubmitPressed}>
        <input
          type="text"
          autoComplete="off"
          className={styles.input}
          maxLength={200}
          placeholder={"Honorific and name"}
          onChange={(event) => setName(event.target.value)}
          value={name}
        />
        <Button>Write To Us</Button>
      </form>
    </div>
  )
}

export default LetterBox
