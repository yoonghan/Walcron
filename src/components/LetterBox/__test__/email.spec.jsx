import * as React from 'react';
import { shallow } from 'enzyme';
import LetterBox from "../index";

describe('LetterBox', () => {
  it('renders normally', () => {
    const wrapper = shallow(<LetterBox/>);
    expect(wrapper.debug()).toMatchSnapshot();
  })
});
