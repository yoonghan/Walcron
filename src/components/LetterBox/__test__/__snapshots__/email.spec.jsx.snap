// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterBox renders normally 1`] = `
"<div>
  If you are interested to talk to us, leave us your contact. Let us reach you instead.
  <NoJavascriptInverse noScriptElem={{...}}>
    <form onSubmit={[Function (anonymous)]} className=\\"jsx-4258014400 letterbox-container\\">
      <input type=\\"text\\" autoComplete=\\"off\\" maxLength={200} placeholder=\\"Honorific and name\\" onChange={[Function (anonymous)]} value=\\"\\" className=\\"jsx-4258014400 letterbox-input\\" />
      <Button onClickCallback={[Function (anonymous)]}>
        Send
      </Button>
      <JSXStyle id=\\"2580228993\\" dynamic={{...}}>
        .letterbox-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-top:20px;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.gap.__jsx-style-dynamic-selector{margin-top:20px;}.letterbox-input.__jsx-style-dynamic-selector{color:#FFF;background-color:transparent;border:1px solid #FFF;margin:0;min-width:200px;display:block;font-size:1rem;box-sizing:content-box;padding:1rem;border-radius:0.3rem;height:1.1rem;text-shadow:0 0 0 #EBEBEB;}
      </JSXStyle>
    </form>
  </NoJavascriptInverse>
</div>"
`;
